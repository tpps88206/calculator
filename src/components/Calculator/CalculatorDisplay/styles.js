const styles = theme => ({
  calculatorDisplay: {
    height: '20%',
    color: 'white',
    fontSize: '6.5em',
    flex: 1,
  },
});

export default styles;
