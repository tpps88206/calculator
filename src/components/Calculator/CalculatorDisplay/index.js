import React from 'react';

import { makeStyles } from '@material-ui/styles';

import AutoScalingText from '../AutoScalingText';
import styles from './styles';

const useStyles = makeStyles(styles);

const CalculatorDisplay = ({ value, ...props }) => {
  const classes = useStyles();

  return (
    <div {...props} className={classes.calculatorDisplay}>
      <AutoScalingText>{value}</AutoScalingText>
    </div>
  );
};

export default CalculatorDisplay;
