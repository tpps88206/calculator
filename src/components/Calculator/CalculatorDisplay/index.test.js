import React from 'react';

import { render } from '../../../utils/test-utils';
import CalculatorDisplay from './index';

it('renders CalculatorDisplay without crashing', () => {
  const div = document.createElement('div');
  render(<CalculatorDisplay />, div);
});

it('renders CalculatorDisplay with snapshot', () => {
  const { asFragment } = render(<CalculatorDisplay />);
  expect(asFragment()).toMatchSnapshot();
});
