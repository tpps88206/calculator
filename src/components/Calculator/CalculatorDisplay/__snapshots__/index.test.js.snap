// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CalculatorDisplay with snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-calculatorDisplay-3"
  >
    <div
      class="makeStyles-autoScalingText-4 position-absolute d-inline-block"
      id="AutoScalingText"
      style="transform: scale(1,1);"
    />
  </div>
</DocumentFragment>
`;
